hazelcast/src/test/java/com/hazelcast/map/nearcache/NearCacheTestSupport.java

```diff
@@ -0,0 +1,289 @@
+import com.hazelcast.core.MapStoreAdapter;
+import com.hazelcast.map.AbstractEntryProcessor;
+import com.hazelcast.map.impl.MapService;
+import com.hazelcast.map.impl.proxy.NearCachedMapProxyImpl;
+import com.hazelcast.map.listener.EntryEvictedListener;
+import com.hazelcast.monitor.NearCacheStats;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.test.AssertTask;
+import com.hazelcast.test.HazelcastTestSupport;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class NearCacheTestSupport extends HazelcastTestSupport {
+
+    protected static final int MAX_CACHE_SIZE = 50000;
+    protected static final int MAX_TTL_SECONDS = 3;
+    protected static final int MAX_IDLE_SECONDS = 1;
+
+    /**
+     * The OS Near Caches evict 20% + 1 of the Near Cache.
+     */
+    protected int getExpectedEvictionCount(int size) {
+        return (int) (size * 0.2) + 1;
+    }
+
+    protected void testNearCacheEviction(IMap<Integer, Integer> map, int size) {
+        int expectedEvictions = getExpectedEvictionCount(size);
+
+        // populate map with an extra entry
+        populateMap(map, size + 1);
+
+        // populate Near Caches
+        populateNearCache(map, size);
+
+        NearCacheStats statsBeforeEviction = getNearCacheStats(map);
+
+        // trigger eviction via fetching the extra entry
+        map.get(size);
+
+        waitForNearCacheEvictions(map, expectedEvictions);
+
+        // we expect (size + the extra entry - the expectedEvictions) entries in the Near Cache
+        int expectedOwnedEntryCount = size + 1 - expectedEvictions;
+
+        NearCacheStats stats = getNearCacheStats(map);
+        assertEquals("got the wrong ownedEntryCount", expectedOwnedEntryCount, stats.getOwnedEntryCount());
+        assertEquals("got the wrong eviction count", expectedEvictions, stats.getEvictions());
+        assertEquals("got the wrong expiration count", 0, stats.getExpirations());
+        assertEquals("we expect the same hits", statsBeforeEviction.getHits(), stats.getHits());
+        assertEquals("we expect the same misses", statsBeforeEviction.getMisses(), stats.getMisses());
+    }
+
+    protected void testNearCacheExpiration(final IMap<Integer, Integer> map, final int size, int expireSeconds) {
+        populateMap(map, size);
+        populateNearCache(map, size);
+
+        assertEquals(format("we expected to have all map entries in the Near Cache (%s)", getNearCacheStats(map)),
+                size, getNearCacheStats(map).getOwnedEntryCount());
+
+        final NearCacheStats statsBeforeExpiration = getNearCacheStats(map);
+
+        sleepSeconds(expireSeconds + 1);
```

Comment:
We have to wait until entries are expired. If we sleep longer, nothing bad can happen.